src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -388,41 +382,27 @@ def _deploy_arm_template_core_unmodified(cmd, resource_group_name, template_file
                        deployment)
 
 
-class JsonCTemplate:
-    def __init__(self, template_string):
-        self.template_string = template_string
-
-
-class JSONSerializer(Serializer):
-    def body(self, data, data_type, **kwargs):
-        if data_type in ('Deployment', 'ScopedDeployment', 'DeploymentWhatIf', 'ScopedDeploymentWhatIf'):
-            # Be sure to pass a DeploymentProperties
-            template = data.properties.template
-            if template:
-                data_as_dict = data.serialize()
-                data_as_dict["properties"]["template"] = JsonCTemplate(template)
-
-                return data_as_dict
-        return super(JSONSerializer, self).body(data, data_type, **kwargs)
```

Comment:
After removing the customized serialization, the information of the ARM template can also be printed in the debug log (only the parameter names and values were printed before)

Before
```
{'properties': {'template': <azure.cli.command_modules.resource.custom.JsonCTemplate object at 0x000002020F2EE970>, 'parameters': {'location': {'value': 'westus'}, 'name': {'value': 'azure-cli-deploy-test-nsg1'}}, 'mode': 'Incremental'}}
```

After
```
{"properties": {"template": "{\r\n    \"$schema\": \"https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#\",\r\n    \"contentVersion\": \"1.0.0.0\",\r\n    \"parameters\": {\r\n      \"location\": {\r\n        \"type\": \"string\",\r\n        \"defaultValue\": \"[resourceGroup().location]\",\r\n        \"metadata\": {\r\n          \"description\": \"Location for the network security group.\\r\\n\"\r\n        }\r\n      },\r\n      \"name\": {\r\n        \"type\": \"string\",\r\n        \"metadata\": {\r\n          \"description\": \"Name of the network security group.\\r\\n\"\r\n        }\r\n      }\r\n    },\r\n    \"variables\": {\r\n    },\r\n    \"resources\": [\r\n      {\r\n        \"type\": \"Microsoft.Network/networkSecurityGroups\",\r\n        \"name\": \"[parameters('name')]\",\r\n        \"apiVersion\": \"2015-06-15\",\r\n        \"location\": \"[parameters('location')]\",\r\n        \"properties\": {\r\n          \"securityRules\": [\r\n          ]\r\n        },\r\n        \"dependsOn\": [ ]\r\n      }\r\n    ],\r\n    \"outputs\": {\r\n      \"NewNSG\": {\r\n        \"type\": \"object\",\r\n        \"value\": \"[reference(parameters('name'))]\"\r\n      }\r\n    }\r\n}", "parameters": {"location": {"value": "westus"}, "name": {"value": "azure-cli-deploy-test-nsg1"}}, "mode": "Incremental"}}
```

Reply:
@alex-frankel Do you think it is acceptable to print the ARM template in the CLI debug log?